openfisca_core/periods.py

```diff
@@ -358,6 +358,41 @@ def __str__(self):
             if fragment is not None
             )
 
+    def __cmp__(self, other):
+        unit_a, start_a, size_a = self
+        unit_b, start_b, size_b = other
+
+        if unit_a != unit_b:
+            unit_weights = {
+                u'day': 1,
+                u'month': 2,
+                u'year': 3,
+                }
+            return cmp(unit_weights[unit_a], unit_weights[unit_b])
+
+        if size_a != size_b:
+            return cmp(size_a, size_b)
+
+        return cmp(start_a, start_b)
+
+    def __lt__(self, other):
```

Comment:
Not an expert about these magic functions, but I thought that when you defined `__cmp__`, all the other ones would by default use  `__cmp__`. It's not the case ?